### DISCUSSION TITLE
Ignore certain categories/tags/accounts

### DISCUSSION THREAD

**Discussion by gccollect** [cioraneanu/firefly-pico#66] [Q&A] [answered]
Been using F3 for years and found out about Pico on r/selfhosted. As I am trying it out for the first time, I got a couple of questions: 
1. In F3, I record mortgage interests as a "withdrawal" in my mortgage liability account as well as mortgage payments going from my current account to my mortgage account. This causes "double counting" on the home page since the mortgage payment already covers both capital and interest. In F3, you can get around this by using reports. Is there a way to customize the home page in Pico to ignore certain categories/tags/accounts for the monthly expenses?
2. Is it possible to have the homepage automatically (or manually like in F3) hide accounts with zero balance? I have almost 40 asset accounts in F3, half of which have zero balance, so they needlessly take up a lot of screen space in the Pico homepage, especially on mobile.

**Comment by cioraneanu** [comment]
Thanks for trying the project and for opening a discussion. 

I am relatively new to Firefly so I apreciate feedback from people with more data onboard 😁

1. The option to filter for the transactions you see on the dashboard sounds really useful. Will implement it soon :D

2. You can already mark an account as hidden from dashboard but i guess it's not fun to do it for 40 entries. To option to autohide 0 could indeed be helpful. 

Side question. Can you tell me how many transactions you have in FF3, and what is the response time for fetching a page of transactions? You can check the time inside Pico, on the transactions list page using the Chrome Dev Tools, Network tab. I am asking because someone reported that for 17k transactions, the request took 9seconds which to me sounded scary 😅

**Reply by avee87** [reply]
> someone reported that for 17k transactions, the request took 9seconds which to me sounded scary

That was me. FYI, after a server upgrade from i5-7500T to i5-12500T the same query now takes just under 4 seconds. Still not ideal but much better than before. I guess, this query is quite CPU-intensive in Firefly.

**Comment by gccollect** [comment]
I have about 10k transactions. The transaction tab times out after 4s unless I filter out the transactions.

**Comment by cioraneanu** [comment, marked as answer]
Can you try setting the env variable `NUXT_PUBLIC_QUERY_TIMEOUT` on the **Pico** docker image to something greater like `9000 ` milliseconds and report the exact time? Also courious what DB you're using for FF3. If Firefly grows slow over time we need to find a way to improve it.

Regarding the 2 sugested features they will be merged into the dev branch in the next few days :)

**Reply by gccollect** [reply]
> Can you try setting the env variable `NUXT_PUBLIC_QUERY_TIMEOUT` on the **Pico** docker image to something greater like `9000 ` milliseconds and report the exact time? 

I increased the timeout and the transaction screen took 10.13s to load.

> Also courious what DB you're using for FF3. If Firefly grows slow over time we need to find a way to improve it.

I am using mariadb:11.2 for FF3. 

> Regarding the 2 sugested features they will be merged into the dev branch in the next few days :)

Thank you for the quick turnaround 👍

**Comment by cioraneanu** [comment]
Implemented the setting to hide accounts with 0 balance and to define which tags, categories and accounts should be excluded from the dashboard (`Settings -> Dashboard preferences`). You can give the `:dev` branch a test drive :)

**Reply by gccollect** [reply]
The option to filter out 0 accounts works. The filtering out of categories works in the "Expenses by categories" section. However, when I click on "Expense" the filtered out categories are still showed. Account filtering doesn't appear to work for either.

On an unrelated note, the total on the dashboard doesn't seem to include the liabilities accounts.

**Reply by cioraneanu** [reply]
Support for negated filters was not there for the "transactions list" page. I've just implemented the first version of it today. You can re-pull the `dev` image to test it out. 

Liability accounts will come later #8

**Reply by gccollect** [reply]
Weird the transaction sums are now all zeros but when I click on them there are transactions
![image](https://github.com/cioraneanu/firefly-pico/assets/55197763/af7e2f05-5338-40df-b1a2-dddcfb5a7df4)

**Reply by cioraneanu** [reply]
I'm going to need more information about your used configs. 
What values have you used for the dashboard exclusion filters? Are the above 0 sums correct or are transactions showed after a click correct?

Something that is worth mentioning is that currently the exclussion mechanism doesn work well with multiple values for any given field(ex more than one tag). It should show correct values on the dashboard but will NOT apply the correct filters when navigation to the transactions-list.

**Reply by gccollect** [reply]
Now the transaction sums are non-zero but categories have disappeared (category list is now empty in settings)
![image](https://github.com/cioraneanu/firefly-pico/assets/55197763/f2a62b92-c4c2-44b2-bb2b-db2c71bb52a3)
